docs: Add to FAQ: When / how to use Linked Ingredients

ChipSkylark37 opened the pull request:
## What this PR does / why we need it:

- Added a new FAQ under the Features & Functionality section that goes over how to add Linked Ingredients to a Recipe. This is a question that is often asked in the Discord. 

EDIT:
- As commented below, 2 more FAQs were added and I've marked it ready for review. 

## Which issue(s) this PR fixes:

None - Just working to fill out some additional FAQs

## Special notes for your reviewer:

ChipSkylark37 committed:
Add to FAQ: When / how to use Linked Ingredients

- Added a new FAQ under the Features & Functionality section that goes over how to add Linked Ingredients to a Recipe. This is a question that is often asked in the Discord. 
- This is not ready to publish yet, and I will continue making some small tweaks on this as I plan to also add a 'tips' section to this.

ChipSkylark37 committed:
Verbiage Changes

- Changed to cleanup language

ChipSkylark37 committed:
Add FAQ for 'Should I use Email? 

- This is a question I've seen be asked a few times on Discord so I wanted to outline why would it be used.

ChipSkylark37 committed:
Adding FAQ: Accessing Mealie Externally

- Using Mealie externally is a huge benefit for its feature set, but comes with its own security considerations. Thats why I opted to have this under the Security & Maintenance section.

ChipSkylark37 committed:
Changing Language on External Access

- Added in note on a potential use case (Shopping Lists)

ChipSkylark37 reviewed (commented):
A total of (3) 'FAQs' have been added and should be ready to be reviewed.

If there are other things that normally come up, let me know and I will do my best to write them up.

ChipSkylark37 committed:
Merge branch 'mealie-next' into patch-2

boc-the-git commented on docs/docs/documentation/getting-started/faq.md:
```diff
@@ -207,6 +236,17 @@
 
 ## Security and Maintenance
 
+??? question "How can I use Mealie externally?"
+
+    ### How can I use Mealie externally?
+
+    By default, Mealie is accessible only on the local network where it is set up. Certain features, such as Shopping Lists, are helpful to access when you are not on your local network. If you want to access Mealie externally, there are several methods with examples listed below, each with its own security considerations. Choose the option that best suits your network's needs. Since every networ ...
+
+    1. Reverse Proxy Tunnel (e.g., Nginx, Cloudflare Zero Tunnel Network Access)
+    2. VPN
+    3. Port Forwarding (Note: Exposing internal devices directly to the internet poses a security risk. This method is not recommended without additional safeguards or security measures in place.)
+
+
```
I'm not sure I'm comfortable with publishing this as is at this time. I'd rather remove it altogether and merge the PR with the rest.

The reason for this is that there's no way to give a definitive statement that is applicable to everyone, given how docker works..
If someone's computer is exposed, Mealie can be exposed.
We can't really control/influence any of that.

boc-the-git replied:
And, for someone who doesn't understand those nuances.. they definitely shouldn't be exposing anything.

ChipSkylark37 replied:
There is a community guide on setting up a reverse proxy. We could instead link to that and say "Proceed with caution as any method to expose Mealie to the internet could put your home network at risk" or something like that?

Kuchenpirat commented:
i would suggest not using titles within indidvidual FAQs. See #4899 for my reasoning. 

If you need to visually section up your text you can either just use bold text or use     
`<p style="font-size: 0.75rem; font-weight: 500;">Creative Title Here</p>`

Kuchenpirat commented on docs/docs/documentation/getting-started/faq.md:
```diff
@@ -61,6 +61,35 @@
+
+    You can optionally link the same ingredient to multiple steps, which is useful for prepping an ingredient in one step and using it in another.
+
+??? question "Should I setup Email / What is Email used for?"
```
I would probably move this to "Technical Consideration" and rename it to "Why setup email?"

Also we need a note in here that these links can be distrirbuted by other means as well.

Kuchenpirat commented on docs/docs/documentation/getting-started/faq.md:
```diff
@@ -61,6 +61,35 @@
     4. Click 'Update'
 
 
+??? question "When would I use Linked Ingredients in a Recipe?"
+
+    ### When would I use Linked Ingredients in a Recipe?
```
```suggestion
??? question "Why Link Ingredients to a Recipe Step?"

    ### Why Link Ingredients to a Recipe Step??
```

ChipSkylark37 committed:
Updaing FAQ for requested changes

- Changed to 'Bold' Headers
- Updated Language on Linked Steps
- Moved Email to Technical Consideration
- Removed Steps and Noted Community Guide & Security Considerations for Exposing Mealie

ChipSkylark37 commented:
Made noted changes / suggestions.

Kuchenpirat commented on docs/docs/documentation/getting-started/faq.md:
```diff
@@ -61,6 +61,23 @@
 
+??? question "Why Link Ingredients to a Recipe Step?"
+
+    **Why Link Ingredients to a Recipe Step?**
+
+    Mealie allows you to link ingredients to specific steps in a recipe, ensuring you know exactly when to add each ingredient during the cooking process.
+
+    #### Link Ingredients to Steps in a Recipe
```
Ah i think my comment was a bit confusing. I did not mean the first title within the card that repeasts the question (this one should be in the table of contents) but every other title within a single faq answer should not be an title to not clutter the table of content. The desired result is to just have section titles and the questions in the TOC

```suggestion
    ### Why Link Ingredients to a Recipe Step?

    Mealie allows you to link ingredients to specific steps in a recipe, ensuring you know exactly when to add each ingredient during the cooking process.

    **Link Ingredients to Steps in a Recipe**
    
```

ChipSkylark37 committed:
Change Header Title / Sub-Section Headers

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

ChipSkylark37 committed:
Reverting Header Change

Reverting Header Change for other 2 FAQ

ChipSkylark37 reviewed (commented):
Reverted change for other 2 headers that do not have 'sub-sections'

Kuchenpirat committed:
Merge branch 'mealie-next' into patch-2

ChipSkylark37 commented:
Thanks @Kuchenpirat
